automated_packaging/prepare_changelog.pl

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/perl
+use lib '/usr/local/bin';
+use common_functions;
+use JSON;
+
+#!usr/bin/perl
+$NEW_VERSION = $ARGV[0];
+$PROJECT = $ARGV[1];
+$EARLIEST_PR_DATE = $ARGV[2];
+
+sub add_changelog_header() {
+
+    # Necessary to write log date
+    my @month_names = qw(January February March April May June July August September October November December);
+    my ( $sec, $min, $hour, $mday, $mon, $year, $wday, $yday, $isdst ) = localtime(time);
+    $year += 1900;
+
+    # Update the changelog file
+    open( CHANGELOG_FILE, ">CHANGELOG.md" ) || die "Can't open the Changelog file !";
+    print CHANGELOG_FILE "### $PROJECT v$NEW_VERSION ($month_names[$mon] $mday, $year) ###\n\n";
+    close(CHANGELOG_FILE);
+}
+
+$github_token = get_and_verify_token();
+
+# Checkout the master branch
+`git checkout master`;
+
+# Now create a new branch based on master
+`git checkout -b $PROJECT-$NEW_VERSION-changelog`;
```

Comment:
This fails if there is a branch with same name. You can add random number or currect time up to seconds to create unique branch name every time (for getting epoch: localtime(time)